src/irr/asset/COBJMeshFileLoader.cpp

```diff
@@ -59,6 +407,15 @@ COBJMeshFileLoader::COBJMeshFileLoader(IAssetManager* _manager) : AssetManager(_
 #ifdef _IRR_DEBUG
 	setDebugName("COBJMeshFileLoader");
 #endif
+    auto vs_nouv_unspec = core::make_smart_refctd_ptr<ICPUShader>(VERT_SHADER_NO_UV);
+    auto vs_uv_unspec = core::make_smart_refctd_ptr<ICPUShader>(VERT_SHADER_UV);
+
+    ICPUSpecializedShader::SInfo specinfo({}, nullptr, "main", ICPUSpecializedShader::ESS_VERTEX);
+    auto vs_nouv = core::make_smart_refctd_ptr<ICPUSpecializedShader>(std::move(vs_nouv_unspec), ICPUSpecializedShader::SInfo(specinfo));
+    auto vs_uv = core::make_smart_refctd_ptr<ICPUSpecializedShader>(std::move(vs_uv_unspec), ICPUSpecializedShader::SInfo(specinfo));
+
+    insertShaderIntoCache(vs_nouv, VERT_SHADER_NO_UV_CACHE_KEY, AssetManager);
+    insertShaderIntoCache(vs_uv, VERT_SHADER_UV_CACHE_KEY, AssetManager);
```

Comment:
why fragment shader is not added to cache yet?

Reply:
or will it be added by MTL?

Reply:
because frag shaders are generated dynamically (see `getShaders()`) but i think it can be static (like VS) with the knowledge that bindings in desc sets doesnt have to be continuous

Reply:
yeah bindings don't have to be continuous, you could do one descset layout for all I care, but then you need to hook up a blank 1x1 texture into the empty slots, so you only need 2 fragment shaders.